crates/rust/src/lib.rs

```diff
@@ -582,6 +586,16 @@ impl WorldGenerator for RustWasm {
 
         let module_name = name.to_snake_case();
         files.push(&format!("{module_name}.rs"), src.as_bytes());
+
+        let remapping_keys = self.opts.with.keys().cloned().collect::<HashSet<String>>();
+
+        let unused_keys = remapping_keys
+            .difference(&self.used_with_opts)
+            .collect::<Vec<&String>>();
+
+        if unused_keys.len() > 0 {
+            panic!("unused remappings provided via `with`: {unused_keys:?}")
```

Comment:
I'm not married to the language in this panic so please feel free to suggest something better.

Reply:
Happy to start here and iterate!

Would you be up for a bit of refactoring though to thread around a `Result` instead of panicking here? That I think would provide a better experience for macro users since they wouldn't have to worry about getting a whole stack trace.

Reply:
Absolutely! I actually started there but reduced to this.